common/src/main/java/tds/common/configuration/RedisClusterConfiguration.java

```diff
@@ -0,0 +1,32 @@
+package tds.common.configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import tds.common.cache.RedisJsonSerializer;
+
+/**
+ * This configuration is responsible for initializing a general RedisTemplate
+ * able to serialize/deserialize generic Objects using the application ObjectMapper.
+ */
+@Configuration
+@ConditionalOnProperty("spring.redis.sentinel.master")
+public class RedisClusterConfiguration {
+
+    @Bean
+    @Primary
+    public RedisTemplate<String, Object> redisTemplate(final RedisConnectionFactory redisConnectionFactory,
+                                                       final ObjectMapper objectMapper) {
+        final RedisTemplate<String, Object> redisTemplate = new RedisTemplate<>();
+        redisTemplate.setConnectionFactory(redisConnectionFactory);
+        redisTemplate.setKeySerializer(new StringRedisSerializer());
+        redisTemplate.setValueSerializer(new RedisJsonSerializer(objectMapper));
+        return redisTemplate;
```

Comment:
This is injecting the standard Application-wide One True ObjectMapper, passing it to the RedisJsonSerializer, who then clones and modifies it to embed class information when serializing objects.

The reason I'm cloning-and-modifying is because our general ObjectMapper usage passes the expected class to the "deserialize" method, so the ObjectMapper knows what to hand back.  In the caching implementation, all we know is that we're de-serializing an Object, so the serialized value needs to contain the specific class information.

I don't want to modify the application-global ObjectMapper to embed class information because other services won't be embedding class information in the serialized values that they're passing to us over the wire. (And they shouldn't.)